credentials/apps/api/accreditor.py

```
@@ -0,0 +1,48 @@
+""" Accreditor class identifies relative issuer."""
+from __future__ import unicode_literals
+
+from credentials.apps.api import exceptions
+from credentials.apps.credentials.issuers import ProgramCertificateIssuer
+
+
+class Accreditor(object):
+    """ Accreditor class identifies credential type and calls corresponding issuer
+    class for generating credential.
+    """
+    def __init__(self, issuers=None):
+        if not issuers:
+            issuers = [ProgramCertificateIssuer()]
+
+        self.issuers = issuers
+        self._create_credential_type_issuer_map()
+
+    def _create_credential_type_issuer_map(self):
+        """Creates a map from credential type slug to a list of credential issuers."""
+        self.credential_type_issuer_map = {}
+        for issuer in self.issuers:
+            slug_value = issuer.issued_credential_type.credential_type_slug
+            if slug_value not in self.credential_type_issuer_map:
```

You need to decide the expected behavior. At the very least the issue should be logged. On the opposite extreme, you could raise an error and halt execution. Don't just hide the error. Never do that.